### ISSUE TITLE
[Enhancement] Improve performance of restoration, especially for larger backups

### ISSUE THREAD

**Issue by shreyas-s-rao** [gardener/etcd-backup-restore#604]
**Enhancement (What you would like to be added):**
Improve multiple aspects of restoration, to allow restoration of large ETCDs without running out of disk space or exceeding ETCD DB quota.

**Motivation (Why is this needed?):**
Large ETCDs lead to large backups, with more number of delta snapshots, larger delta snapshots or both. Trying to restore such large ETCDs might fail due to (a) running out of disk space due to storing too many temp delta snapshot files on disk until restoration is complete, (b) embedded etcd DB running out of space during restoration. 

**Approach/Hint to the implement solution (optional):**
- [x] #603 
- [x] Instead of cleaning up the temp files *after* restoration succeeds, cleanup each temp delta snapshot file as and when the delta snapshot is successfully applied to the embedded etcd
- [ ] Periodically run (compaction and) defragmentation on the embedded etcd during restoration, either after every X deltas applied or after every Y GB of etcd DB size reached
- [x] Decompress download delta snapshots just before applying them, thus saving disk space when persisting delta snapshots locally
- [x] Make the temp dir configurable via CLI, since it is currently hardcoded to `/tmp`, in order to decouple from the /tmp folder which might be re-purposed by the OS at any time in the future
- [x] Name the temp files after delta snapshot file names to make debugging issues with cleanup easier
- [x] Improve logging for restoration - ex: log total number of deltas before starting restoring

**Comment by ishan16696** [comment]
backup-restore [v0.22.0](https://github.com/gardener/etcd-backup-restore/releases/tag/v0.22.0) already has been released, @shreyas-s-rao you have wrongly added `v0.22.0` as milestone. so I'm updating the milestone to `v0.23.0`. Please feel free to update milestone further

**Comment by shreyas-s-rao** [comment]
Oops, my bad. Thanks for correcting it.

**Comment by shreyas-s-rao** [comment]
> Periodically run (compaction and) defragmentation on the embedded etcd during restoration, either after every X deltas applied or after every Y GB of etcd DB size reached

This task will be handled in a separate issue (probably https://github.com/gardener/etcd-druid/issues/232 once it moves to etcd-backup-restore repo).
